crates/node/service/src/actors/engine/actor.rs

```diff
@@ -30,30 +30,44 @@ use crate::{NodeActor, actors::CancellableContext};
 pub struct EngineActor {
     /// The [`EngineActorState`] used to build the actor.
     builder: EngineBuilder,
-    /// The receiver for reset requests.
+    /// A channel to receive [`OpAttributesWithParent`] from the derivation actor.
+    attributes_rx: mpsc::Receiver<OpAttributesWithParent>,
+    /// A channel to receive [`OpExecutionPayloadEnvelope`] from the network actor.
+    unsafe_block_rx: mpsc::Receiver<OpExecutionPayloadEnvelope>,
+    /// A channel to receive reset requests.
     reset_request_rx: mpsc::Receiver<()>,
-    /// The sender for L2 safe head update notifications.
-    engine_l2_safe_head_tx: watch::Sender<L2BlockInfo>,
-    /// A channel to send a signal that EL sync has completed. Informs the derivation actor to
-    /// start. Because the EL sync state machine within [`InnerEngineState`] can only complete
-    /// once, this channel is consumed after the first successful send. Future cases where EL
-    /// sync is re-triggered can occur, but we will not block derivation on it.
-    sync_complete_tx: oneshot::Sender<()>,
-    /// A way for the engine actor to send a [`Signal`] back to the derivation actor.
-    derivation_signal_tx: mpsc::Sender<Signal>,
+    /// Handler for inbound queries to the engine.
+    inbound_queries: mpsc::Receiver<EngineQueries>,
+    /// A channel to receive [`RuntimeConfig`] from the runtime actor.
+    runtime_config_rx: mpsc::Receiver<RuntimeConfig>,
+    /// A channel to receive build requests from the sequencer actor.
+    /// TODO(@theochap): plug it in the engine actor.
+    #[allow(dead_code)]
+    build_request_rx:
+        mpsc::Receiver<(OpAttributesWithParent, mpsc::Sender<OpExecutionPayloadEnvelope>)>,
+    /// The [`L2Finalizer`], used to finalize L2 blocks.
+    finalizer: L2Finalizer,
 }
 
 /// The outbound data for the [`EngineActor`].
 #[derive(Debug)]
-pub struct EngineOutboundData {
-    /// A channel to request the engine actor to reset.
+pub struct EngineInboundData {
+    /// A channel to send [`OpAttributesWithParent`] and [`OpExecutionPayloadEnvelope`] to the
+    /// sequencer actor.
```

Comment:
Comment is a bit inaccurate. This channel is to send attributes for the engine to build, and a channel to return the built payload back.